packages/p2p/src/components/modal-manager/modals/rate-change-modal/rate-change-modal.jsx

```diff
@@ -1,15 +1,19 @@
-import { observer } from 'mobx-react-lite';
 import React from 'react';
 import { Button, Modal, Text } from '@deriv/components';
 import { isMobile } from '@deriv/shared';
+import { observer, useStore } from '@deriv/stores';
 import { localize, Localize } from 'Components/i18next';
 import { useStores } from 'Stores';
 import { useModalManagerContext } from 'Components/modal-manager/modal-manager-context';
 import './rate-change-modal.scss';
 
 const RateChangeModal = ({ currency }) => {
-    const { floating_rate_store, general_store } = useStores();
-    const local_currency = currency ?? general_store.client?.local_currency_config?.currency;
+    const {
+        client: { local_currency_config },
+    } = useStore();
+
+    const { floating_rate_store } = useStores();
```

Comment:
```suggestion
    const { floating_rate_store } = useP2PStores();
```
@ameerul-deriv Just a suggestion, Maybe can be done in a different PR, To avoid confusion we can rename this hook, Same as we did in cashier so now we have `useCashierStores` hook.

Reply:
ah yeah carol had mentioned she wanted to do this, so we will be doing this in the future!